sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/pkversion/LeaseStoreManagerImpl.java

```diff
@@ -203,6 +207,29 @@ public Mono<Void> delete(Lease lease) {
             .then();
     }
 
+    @Override
+    public Mono<Void> delete(List<Lease> leases) {
+        checkNotNull(leases, "Argument 'leases' can not be null");
+
+        List<CosmosItemOperation> operations = new ArrayList<>();
+        for (Lease lease : leases) {
+            operations.add(CosmosBulkOperations.getDeleteItemOperation(lease.getId(), new PartitionKey(lease.getId())));
+        }
+
+        return this.leaseDocumentClient.executeBulkOperations(Flux.defer(() -> Flux.fromIterable(operations)))
+            .flatMap(itemResponse -> {
+                if (itemResponse.getResponse() != null && itemResponse.getResponse().isSuccessStatusCode()) {
```

Comment:
Can't that run into a hnag very easily. delete being called concurrently could result in operations being added for docs that might not exist anymore - 404 would not satisfy isSuccessStatusCode and you'll end-up in an endless loop in operations.size != 0?

Reply:
you are right Fabian, I need to add more handling around it -- want to get some feedback around the delete leases part in the meeting today